src/hatch/utils/metadata.py

```diff
@@ -0,0 +1,8 @@
+from __future__ import annotations
+
+import re
+
+
+def normalize(name: str) -> str:
```

Comment:
I think this is too generic a name and would cause a bit of friction when reading code that uses it.

Reply:
normalize_names or normalize_package_names?